app.rb

```diff
@@ -0,0 +1,157 @@
+require_relative 'student'
+require_relative 'teacher'
+require_relative 'book'
+require_relative 'rental'
+require_relative 'classroom'
+
+class App
+  def initialize
+    @books = []
+    @person = []
+    @rentals = []
+  end
+
+  # rubocop:disable Style/CyclomaticComplexity
+  def select_opt
+    option = gets.chomp.to_i
+    case option
+    when 1 then list_books
+    when 2 then list_people
+    when 3 then create_person
+    when 4 then create_book
+    when 5 then create_rental
+    when 6 then list_rentals
+    when 7 then exit(true)
+    else
+      puts 'PLEASE ENTER A NUMBER (1..7)'
+    end
+  end
+  # rubocop:enable Style/CyclomaticComplexity
+
+  def list_books
+    @books.each do |book|
+      puts "Title: #{book.title}, Author: #{book.author}"
+    end
+  end
+
+  def list_people
+    @person.each do |p|
+      puts "[#{p.class.name}] Name: #{p.name}, ID: #{p.id}, Age: #{p.age}"
+    end
+  end
+
+  def create_book
+    print 'Title:'
+    title = gets.chomp
+    print 'author:'
+    author = gets.chomp
+    @books.push(Book.new(title, author))
+    puts 'Book Created Successfully'
+  end
+
+  def check_number(msg)
+    number = 0
+    loop do
+      print msg
+      input = gets.chomp.to_i
+      if input.is_a?(Integer) && input.positive?
+        number = input
+        break
+      else
+        puts 'Please, enter a valid input!'
+      end
+    end
+    number
+  end
+
+  def check_options(msg, options)
+    number = 0
+    loop do
+      print msg
+      input = gets.chomp.to_i
+      if options.include?(input)
+        number = input
+        break
+      else
+        puts 'Please, enter a valid input!'
+      end
+    end
+    number
+  end
+
+  def check_permission(permission)
+    case permission
+    when 'y' then permission = true
+    when 'n' then permission = false
+    end
+    permission
+  end
+
+  def create_student(name, age, classroom)
+    print 'Has parent permission? [y/n]:'
+    permission = gets.chomp
+    parent_permission = check_permission(permission)
+    student_classroom = Classroom.new(classroom)
+    student = Student.new(age, name, parent_permission: parent_permission)
+    student.classroom = student_classroom
+    @person.push(student)
+  end
```

Comment:
- [ ] I got an error when I wanted to create a student. This is because the app asks for more than the required number of parameters. Let's remember that asking for a classroom is not necessary when creating a person.

![student](https://user-images.githubusercontent.com/53124548/216763891-e25a1988-5fc2-4d3b-a8cf-080c2f6e016f.png)